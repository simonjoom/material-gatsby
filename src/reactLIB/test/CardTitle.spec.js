import React from 'react';
import { shallow } from 'enzyme';
import CardTitle from '../src/CardTitle';

describe('<CardTitle />', () => {
  let wrapper;

  test('renders', () => {
    wrapper = shallow(<CardTitle image="">I am a very simple card</CardTitle>);
    expect(wrapper).toMatchSnapshot();
  });

  test('requires an image prop', () => {
    wrapper = shallow(
      <CardTitle image="url.jpg" reveal>
        I am a very simple card
      </CardTitle>
    );

    expect(wrapper.find('img').hasClass('activator')).toBeTruthy();
  });

  test('can have waves', () => {
    wrapper = shallow(
      <CardTitle image="url.jpg" waves="green">
        I am a very simple card
      </CardTitle>
    );

    expect(wrapper.hasClass('waves-green')).toBeTruthy();
  });

  test('accepts classnames', () => {
    wrapper = shallow(
      <CardTitle image="url.jpg" waves="green" className="find">
        I am a very simple card
      </CardTitle>
    );

    expect(wrapper.hasClass('find')).toBeTruthy();
  });
});
