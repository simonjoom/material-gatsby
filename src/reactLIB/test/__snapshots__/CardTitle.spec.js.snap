// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardTitle /> renders 1`] = `
<div
  className="card-image"
>
  <img
    className=""
    src=""
  />
  <span
    className="card-title"
  >
    I am a very simple card
  </span>
</div>
`;
